tests/cases/viewport_depth.amber

```diff
@@ -0,0 +1,70 @@
+#!amber
+#
+# Copyright 2021 The Amber Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SHADER vertex vert_shader GLSL
+#version 430
+
+layout(location = 0) in vec4 position;
+
+void main() {
+  gl_Position = vec4(position.xy, 0.5, 1.0);
+}
+END
+
+SHADER fragment frag_shader GLSL
+#version 430
+
+layout(location = 0) out vec4 final_color;
+
+void main() {
+  final_color = vec4(0, 1, 0, 1);
+}
+END
+
+BUFFER position DATA_TYPE vec2<float> DATA
+-1.0 -1.0
+ 1.0 -1.0
+ 1.0  1.0
+-1.0  1.0
+END
+
+BUFFER framebuffer FORMAT B8G8R8A8_UNORM
+BUFFER depth FORMAT D32_SFLOAT
+
+PIPELINE graphics pipeline1
+  ATTACH vert_shader
+  ATTACH frag_shader
+
+  DEPTH
+    TEST on
+    WRITE on
+  END
+
+  VERTEX_DATA position LOCATION 0
+
+  FRAMEBUFFER_SIZE 256 256
+  VIEWPORT 0.0 0.0 SIZE 256.0 256.0 MIN_DEPTH 0.3 MAX_DEPTH 0.9
+
+  BIND BUFFER framebuffer AS color LOCATION 0
+  BIND BUFFER depth AS depth_stencil
+END
+
+CLEAR_DEPTH pipeline1 1.0
+CLEAR_COLOR pipeline1 255 255 255 255
+CLEAR pipeline1
+RUN pipeline1 DRAW_ARRAY AS TRIANGLE_FAN START_IDX 0 COUNT 4
+
+EXPECT depth IDX 0 TOLERANCE 1.0e-6 EQ 0.6
```

Comment:
I know it's not possible to probe different locations of a depth buffer, but you could have a separate pipeline that converts the depth data to a grayscale format that you then could use. There are some amber files in the same directory that does that, so you can use one as a reference. Then you could have the viewport size something else than the whole frame buffer.

Reply:
I would be happy for this to come as a followup PR (perhaps even you, Ari, can take care of this?), as this sounds fairly unintuitive.

Reply:
I looked at this a little bit because the tests that I was originally trying to create are for depth-related things. Since rendering to floating-point images is relatively common (and is particularly useful for testing), could we instead create a syntax for verifying floating-point images as images? I haven't really looked into this, so I don't know how complex it would be, but having to add extra pipelines to reprocess any image with a bit-depth over 8 seems a bit clunky to me.

As for making this test use a viewport other than the whole framebuffer, this was actually my original usecase, just setting a depth range for the viewport, so that's where this test came from. Certainly it could be made to use a different viewport setting once the verification is changed.